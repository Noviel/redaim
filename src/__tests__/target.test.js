import createTarget from '../target';

describe('createTarget', () => {
  it('should return correct object with default properties', () => {
    const result = createTarget()();

    const toMatch = {
      initialState: result.hasOwnProperty('initialState'),
      reducer: result.hasOwnProperty('reducer')
    };

    expect(toMatch).toMatchSnapshot();
  });

  it('should set initialState', () => {
    const result = createTarget()({ value: 'initialValue' });

    expect(result.initialState).toMatchSnapshot();
  });

  it('should set reducer', () => {
    const { reducer } = createTarget({
      'ACTION': (state, action) => state + 1
    })();

    const toMatch = {
      state: reducer(100, { meta: { exec: 'ACTION' } })
    };
    
    expect(toMatch).toMatchSnapshot();
  });  
});
