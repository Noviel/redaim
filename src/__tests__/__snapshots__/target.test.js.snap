// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createTarget should return correct object with default properties 1`] = `
Object {
  "initialState": true,
  "reducer": true,
}
`;

exports[`createTarget should set initialState 1`] = `
Object {
  "value": "initialValue",
}
`;

exports[`createTarget should set reducer 1`] = `
Object {
  "state": 101,
}
`;
